modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -715,13 +715,16 @@ public MediaPackage addTrack(InputStream in, String fileName, MediaPackageElemen
       job = serviceRegistry.updateJob(job);
       String elementId = UUID.randomUUID().toString();
       logger.info("Start adding track {} from input stream on mediapackage {}", elementId, mediaPackage);
+      if (fileName.length() > 75) {
+        fileName = fileName.substring(0, 75) + '.' + FilenameUtils.getExtension(fileName);
```

Comment:
1. Consider marking altered file names as such in case the new name gets offensive.
2. Consider declaring 75 as a constant.
3. If the old file name must not be longer than 75 characters (checked in if condition), the resulting file name probably should also not be longer (currently it's 75 + 1 + file extension length)

```suggestion
        String fileExtension = FilenameUtils.getExtension(fileName);
        String separator = '_lop.';
        int baseNameLength = 75 - fileExtension.length() - separator.length();
        fileName = fileName.substring(0, baseNameLength) + separator + fileExtension;
```

Reply:
We don't have to hit exactly 75 characters, do we? I'm in favor of keeping things less complicated. I also don't understand your `_lop.` and I'm not sure others will. What about just using the default setting of [`StringUtils.abbreviate(…`)](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#abbreviate-java.lang.String-int-) instead?

Reply:
The current implementation will create file names like `Some_File.ex.ext` from `Some_File.ext`, assuming that `Some_File.ext` is 76 characters long.

`StringUtils.abbreviate(…)` looks like a good default but will produce many dots just before the file name extension (and some ill-configured snakeoil virus protection may have an issue with that). I have no problem causing false snakeoil alerts, though.

Reply:
Catching half of the extension is actually a good point. That could look weird. I'll update the code.